### PR TITLE
Issue #150 Use docker-in-docker instead of mounting host docker

### PR DESCRIPTION
**Pull request by frodopwns**
Initial implementation of dind sidecar.

### CODE REVIEW

File: chart/templates/deployment.yaml [hunk 1]
```diff
@@ -21,6 +21,7 @@ spec:
         - --registry-url={{ .Values.registry.url }}
         - --registry-org={{ .Values.registry.org }}
         - --registry-auth={{ .Values.registry.authtoken }}
+        - --docker-from-env=true
```
Comment (frodopwns) [inline chart/templates/deployment.yaml]: I'm assuming we want to use a value from values.yaml for this?

File: chart/templates/deployment.yaml [hunk 2]
```diff
@@ -39,12 +42,19 @@ spec:
           httpGet:
             path: /ping
             port: {{ .Values.service.http.internalPort }}
-        volumeMounts:
-        - mountPath: /var/run/docker.sock
-          name: docker-socket
+      - name: dind-daemon 
+        image: docker:dind 
+        resources:
```
Comment (frodopwns) [inline chart/templates/deployment.yaml]: These were just carried over from an example. Anyone have input on what this should be set to?

Comment (bacongobbler) [inline chart/templates/deployment.yaml]: we can just remove this for now. This **may** be why you were seeing performance hits with this PR. I'll give this a spin with and without the resource constraints. :)

File: chart/templates/deployment.yaml (hunk 1 above)
Comment (bacongobbler) [inline chart/templates/deployment.yaml]: Good point. I think we can actually remove this flag from the deployment and just [flip the default flag value](https://github.com/Azure/draft/blob/master/cmd/draftd/start.go#L64) to `true`. Would you mind doing that instead? :)

Comment (frodopwns) [inline chart/templates/deployment.yaml]: Sure thing!

**Comment by bacongobbler** [comment]
this worked beautifully for me, and even enabled me to build apps utilizing new docker builder features like #142. This is fantastic stuff, so thank you!

**Comment by bacongobbler** [comment]
oops, actually I spoke a little too soon... I just saw this after building a few apps:

```
Error encountered streaming JSON response: mkdir /var/lib/docker/vfs/dir/394acd1889a088fbd52717a4021c5abe9436c24d1ac529be745f7e7615e6cbce-init: no space left on device
```

I'll do some investigation on the emptydir volume and see if we can expand the size somehow.

**Comment by bacongobbler** [comment]
```
/ # df -h
Filesystem                Size      Used Available Use% Mounted on
overlay                  16.1G     16.1G         0 100% /
tmpfs                  1000.2M         0   1000.2M   0% /dev
tmpfs                  1000.2M         0   1000.2M   0% /sys/fs/cgroup
/dev/sda1                16.1G     16.1G         0 100% /dev/termination-log
/dev/sda1                16.1G     16.1G         0 100% /var/lib/docker
tmpfs                  1000.2M     12.0K   1000.2M   0% /var/run/secrets/kubernetes.io/serviceaccount
/dev/sda1                16.1G     16.1G         0 100% /etc/resolv.conf
/dev/sda1                16.1G     16.1G         0 100% /etc/hostname
/dev/sda1                16.1G     16.1G         0 100% /etc/hosts
shm                      64.0M         0     64.0M   0% /dev/shm
none                   1000.2M         0   1000.2M   0% /tmp
/ # docker ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
/ # docker images -a
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
<none>              <none>              bbd60477b5be        2 minutes ago       694MB
<none>              <none>              cb1768ff9389        3 minutes ago       694MB
<none>              <none>              946a1ee5eb32        4 minutes ago       694MB
<none>              <none>              347ce62e498b        5 minutes ago       684MB
python              onbuild             9d2405932d52        3 days ago          684MB
maven               3.5-jdk-8-alpine    fc040281fd6f        5 days ago          116MB
```

Let me try to delete this minikube cluster and start from a fresh slate. I bet this is just a client-side issue that I need to expand my minikube storage.

**Comment by bacongobbler** [comment]
Yep. New cluster fixed this.

```
><> kk exec draftd-1168773677-09c3g -c dind-daemon -- df -h
Filesystem                Size      Used Available Use% Mounted on
overlay                  16.1G      2.0G     13.2G  13% /
```

Thanks!

**Pull request merged**